src/lightning/pytorch/loggers/neptune.py

```diff
@@ -39,8 +39,8 @@
 log = logging.getLogger(__name__)
 
 # neptune is available with two names on PyPI : `neptune` and `neptune-client`
-_NEPTUNE_AVAILABLE = RequirementCache("neptune>=1.0")
-_NEPTUNE_CLIENT_AVAILABLE = RequirementCache("neptune-client")
+_NEPTUNE_AVAILABLE = bool(RequirementCache("neptune>=1.0")) or bool(RequirementCache("neptune-client>=1.0"))
+_LEGACY_NEPTUNE_CLIENT_AVAILABLE = bool(RequirementCache("neptune-client<1.0"))
```

Comment:
These shouldn't be converted to bools. This class overrides __str__ providing a helpful error message to the users. See for instance the use in L227

Reply:
```suggestion
_NEPTUNE_AVAILABLE = RequirementCache("neptune>=1.0") or RequirementCache("neptune-client>=1.0")
_LEGACY_NEPTUNE_CLIENT_AVAILABLE = RequirementCache("neptune-client<1.0")
```

Reply:
@Raalsky What's the issue with having two separate variables each pointing to each class? (as in the current version)

Reply:
> These shouldn't be converted to bools. This class overrides str providing a helpful error message to the users. See for instance the use in L227

I've been trying during the development and I remember that this leads to two separate messages about the lack of one of the packages. The expected behavior is to raise/warn whenever none of `neptune-client`/ `neptune` was installed. This should be covered in: https://github.com/Lightning-AI/pytorch-lightning/pull/19131/files#diff-903c2ecc76f95d71b3f577baf15c47f0e629f6e2a4b201a344177e7908cfccd9R323

> What's the issue with having two separate variables each pointing to each class? (as in the current version)

`neptune` was introduced as a name transition of `neptune-client` and the long-term target is to get rid of `Neptune-client` package completely someday. It was introduced as a part of breaking-changes with a release of `neptune-client==1.0`. `neptune-client>=1.0` is just an alias of `neptune` package and have some breaking-changes in compare to `neptune-client<1.0.0`. We would like to still support all of the possible environments and package versions. The earlier approach was just "wrong" and had some misunderstandings of how the relationships between each of the packages looked like.

Reply:
If that's the motivation, then this should work (see other suggestions too=:

```python
_NEPTUNE_AVAILABLE = RequirementCache("neptune>=1.0")
# neptune-client will be gone someday. additionally, version 1.0 had breaking changes to account for
_NEPTUNE_CLIENT_AVAILABLE = RequirementCache("neptune-client>=1.0")
_LEGACY_NEPTUNE_CLIENT_AVAILABLE = RequirementCache("neptune-client<1.0")
_NEPTUNE_GREATER_EQUAL_1_0 = bool(_NEPTUNE_AVAILABLE or _NEPTUNE_CLIENT_AVAILABLE)
```